request-attrs.go

```diff
@@ -3,29 +3,29 @@ package sftp
 // Methods on the Request object to make working with the Flags bitmasks and
 // Attr(ibutes) byte blob easier. Use Pflags() when working with an Open/Write
 // request and AttrFlags() and Attributes() when working with SetStat requests.
-import "os"
+import (
+	"os"
+
+	sshfx "github.com/pkg/sftp/internal/encoding/ssh/filexfer"
+)
 
 // FileOpenFlags defines Open and Write Flags. Correlate directly with with os.OpenFile flags
 // (https://golang.org/pkg/os/#pkg-constants).
 type FileOpenFlags struct {
 	Read, Write, Append, Creat, Trunc, Excl bool
 }
 
-func newFileOpenFlags(flags uint32) FileOpenFlags {
-	return FileOpenFlags{
-		Read:   flags&sshFxfRead != 0,
-		Write:  flags&sshFxfWrite != 0,
-		Append: flags&sshFxfAppend != 0,
-		Creat:  flags&sshFxfCreat != 0,
-		Trunc:  flags&sshFxfTrunc != 0,
-		Excl:   flags&sshFxfExcl != 0,
-	}
-}
```

Comment:
The tests on this functionality was being built on these helper functions, rather than on the `Request` object receiver methods themselves. I moved these into the receiver methods and fixed the tests to use example Request objects with the specific appropriate fields set. So, now the tests have a bit more coverage, and are less “synthetic”.